packages/core/src/util/txInspector.ts

```diff
@@ -198,17 +219,69 @@ export const stakeKeyDeregistrationInspector: StakeKeyRegistrationInspector = (t
  * @param {TxAlonzo} tx transaction to inspect
  * @returns {WithdrawalInspection} accumulated withdrawal quantities
  */
-export const withdrawalInspector: WithdrawalInspector = (tx: TxAlonzo) =>
+export const withdrawalInspector: WithdrawalInspector = (tx) =>
   tx.body.withdrawals?.length ? BigIntMath.sum(tx.body.withdrawals.map(({ quantity }) => quantity)) : 0n;
 
+/**
+ * Inspects a transaction for minted or burned assets.
+ *
+ * @returns {Value} A collection of minted and burned assets grouped by policy Id.
+ */
+export const assetsMintedInspector: AssetsMintedInspector = (tx) => {
+  const assets: AssetsMintedInspection = { burned: [], minted: [] };
+  const scriptMap = new Map();
+
+  if (!tx.body.mint) return assets;
+
+  // Scripts can be embedded in transaction auxiliary data and/or the transaction witness set. If this transaction
+  // was built by this client the script will be present in the witness set, however, if this transaction was
+  // queried from a remote repository that doesn't fetch the witness data of the transaction we can still check
+  // if the script is present in the auxiliary data.
+  const scripts = [...(tx.auxiliaryData?.body?.scripts || []), ...(tx.witness?.scripts || [])];
+
+  for (const script of scripts) {
+    let policyId;
+    switch (script.__type) {
+      case ScriptType.Native:
+        policyId = util.nativeScriptPolicyId(script);
```

Comment:
I think you can declare policyId inside the case block:
```diff
-   let policyId;
    switch (script.__type) {
      case ScriptType.Native:
-       policyId = util.nativeScriptPolicyId(script);
+       const policyId = util.nativeScriptPolicyId(script);

Reply:
This violates the following lint rule: 
https://eslint.org/docs/latest/rules/no-case-declarations

Reply:
after re-reading the rule carefully, this is only a violation if the all case blocks share the same scope, so I added a separate scope for the case statement.

addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/447/commits/d781ee7b707c4788b53cc4894363002ad026ba10